spark/src/test/java/org/opensearch/sql/spark/asyncquery/IndexQuerySpecTest.java

```diff
@@ -549,76 +492,55 @@ public GetJobRunResult getJobRunResult(String applicationId, String jobId) {
                       return null;
                     }
                   };
-              EMRServerlessClientFactory emrServerlessClientFactory =
-                  new EMRServerlessClientFactory() {
-                    @Override
-                    public EMRServerlessClient getClient() {
-                      return emrsClient;
-                    }
-                  };
+              EMRServerlessClientFactory emrServerlessClientFactory = () -> emrsClient;
               AsyncQueryExecutorService asyncQueryExecutorService =
                   createAsyncQueryExecutorService(emrServerlessClientFactory);
 
               // Mock flint index
               mockDS.createIndex();
               // Mock index state
-              MockFlintSparkJob flintIndexJob = new MockFlintSparkJob(mockDS.latestId);
+              MockFlintSparkJob flintIndexJob =
+                  new MockFlintSparkJob(stateStore, mockDS.latestId, DATASOURCE);
               flintIndexJob.deleting();
 
               // 1. drop index
               CreateAsyncQueryResponse response =
                   asyncQueryExecutorService.createAsyncQuery(
                       new CreateAsyncQueryRequest(mockDS.query, DATASOURCE, LangType.SQL, null));
 
+              AsyncQueryExecutionResponse asyncQueryExecutionResponse =
+                  asyncQueryExecutorService.getAsyncQueryResults(response.getQueryId());
               // 2. fetch result
+              assertEquals("FAILED", asyncQueryExecutionResponse.getStatus());
               assertEquals(
-                  "SUCCESS",
-                  asyncQueryExecutorService
-                      .getAsyncQueryResults(response.getQueryId())
-                      .getStatus());
-
-              flintIndexJob.assertState(FlintIndexState.DELETED);
+                  "Transaction failed as flint index is not in a valid state.",
+                  asyncQueryExecutionResponse.getError());
+              flintIndexJob.assertState(FlintIndexState.DELETING);
             });
   }
 
-  /**
-   * No Job running, expectation is
-   *
-   * <p>(1) not call EMR-S (2) change index state to: DELETED
-   */
+  /** No Job running, expectation is */
```

Comment:
why ignore emr-s call verification?

Reply:
So for refreshing, empty, active and creating states. we cancel the job and removed emrs job verification. Let me add call asserts.

For other transitioning states, the query will result in error and in that case we are asserting no emr calls.
Let me know your thoughts on this.